### DISCUSSION TITLE
MSSQL Network-related or instance error

### DISCUSSION THREAD

**Discussion by adriantheking** [AzureCommunityPL/group#362] [Pomoc] [answered]
Hej,

Mam następującą sytuację: 
Durable function wykonuje założony proces i jednym z jego zadań pod koniec jest zapisanie wyników w bazie danych hostowanej na Azurze. Ilość danych zapisywanych to dzięsiątki tysięcy rekordów, oczywiście wszystko idzie poprzez bulk insert w procedurze którą wywołuje dapper. Tutaj pojawia się od czasu do czasu problem gdy równocześnie uruchomionych jest kilkanaście/kilkadziesiąt orkiestratorów. Średnio 2-3 z 20 instancji zwróci mi error: 

`"Activity function '[moja az func zapisująca dane]' failed: A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 0 - A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.)"`

Do nawiązania połączenia z bazą danych i uzyskania tokenu używam klasy z Azure.Identity: DefaultAzureCredential()

Chwilowo zaczyna mi brakować pomysłów gdzie powinienem szukać problemu.

**Comment by krukowskid** [comment]
Jaki masz tier tej bazy? Wygląda trochę jakbyś wysycał limit dostępnych zasobów

**Reply by adriantheking** [reply]
Service tier: General Purpose (Scalable compute and storage options)
Compute tier: Serverless

Skalowanie ustawione do 2 vCores

**Reply by kamil-mrzyglod** [reply]
A to nie jest tak, że dla niektórych instancji walisz w warmup serverlessowej bazy i stąd błędy, że nie może znaleźć serwera? Masz jakieś retry policy zaimplementowane?

**Reply by adriantheking** [reply]
Raczej problem jest gdzieś indziej, na samym początku zaciągana jest konfiguracja z bazy i przechodzi wszystko bez problemu, co do retry policy to  mam do innej sytuacji, może dopisanie tego będzie też częściowym rozwiązaniem ;)

**Reply by kamil-mrzyglod** [reply]
A ten config początkowy jak jest pobierany? To jest jedna z activities / element całej orkiestracji? Teoretycznie może być tak jak pisał @krukowskid że wysycasz jakiś limit. Połączenie z bazą idzie przez jakiegoś singletona czy robisz connection per orkiestracja / activity?

**Reply by adriantheking** [reply]
Pobieranie konfiguracji / zapisywanie danych itd są osobnymi activities. Połączenie w startupie zarejestrowane jako Scoped.

**Comment by adriantheking** [comment, marked as answer]
Po dyskusji z supportem, namierzyliśmy problem. Podczas wykonywania kroków w workflow czasami zajmują wiecej niż 30min, workflow wtedy jest usypiany, a podczas "ożywienia" połączenie jest zrywane ze względu na zbyt długi czas idle.
